media/plg_system_tour/js/guide.js

```diff
@@ -0,0 +1,92 @@
+Joomla = window.Joomla || {};
+
+(function (Joomla, window) {
+    document.addEventListener('DOMContentLoaded', function() {
+        var urlParameters = Joomla.getOptions('tour-guide');
+        if (!urlParameters) { return; }
+        if (window.location.href.indexOf("&") > -1) {
+            var filename = '';
+            var URL = '';
+            //case if their is presence of two paramteres and one of them is option
+            if (urlParameters.urlOption) {
+                filename = urlParameters.urlOption;
+
+                //case if their is presence of two paramteres and they are view and option
+                if (urlParameters.urlView) {
+
+                    filename = filename + '_' + urlParameters.urlView;
+                }
+
+                //case if their is presence of three paramteres and they are view and option and layout
+                if ((window.location.href.indexOf("layout=") > 0)) {
+
+                    filename = filename + '_' + urlParameters.urlLayout;
+                }
+                //made the file name to be fetched out
+                filename = filename + '.' + 'json';
+                // path from where the file to be fetched
+                URL = urlParameters.urlOption + '/' + filename;
+            }
+        } else {
+            filename = urlParameters.urlOption + '.' + 'json';
+            URL = urlParameters.urlOption + '/' + filename;
+        }
+        var btn= document.createElement('button');
+        btn.classList.add('btn');
+        btn.classList.add('btn-sm');
+        btn.classList.add('btn-outline-primary');
+        btn.setAttribute('id', 'startTourBtn');
+        btn.innerHTML = '<span class="fa fa-map-signs" aria-hidden="true"></span>'+urlParameters.btnname+'</button>';
+        document.getElementById('toolbar').appendChild(btn);
+        Joomla.request({
+            url: 'https://yveshoppe.de/jdocsapi/' +  urlParameters.langtag + '/' + filename,
```

Comment:
Put 'https://yveshoppe.de/jdocsapi/' in a variable, so we can easily adjust it (you have it 3 times in the code). Actually we need to think on a way to completely customize it depending on the extension that is shown (so 3rd parties can use it).